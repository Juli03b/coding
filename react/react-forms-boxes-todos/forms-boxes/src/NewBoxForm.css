form{
    text-align: -webkit-center;
}
label{
    top: 20px;
    display: block;
    font-size: 100px;
}
input{
    height: 20px;
    width: 200px;
}
button{
    padding: 2.5px;
}