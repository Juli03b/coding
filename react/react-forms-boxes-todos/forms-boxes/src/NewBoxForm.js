import { useState } from "react";
import "./NewBoxForm.css"

const NewBoxForm = ({submitBox}) => {
    const INITIAL_STATE = {color: "", height: "", width: ""};
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData( formData => ({
            ...formData,
            [name]: value
        }));
    }
    const handleSubmit = (evt) => {
        evt.preventDefault();
        submitBox(formData);
        setFormData(INITIAL_STATE);
    }
    
    return (
        <form onSubmit={handleSubmit} data-testid="form">
            <label htmlFor="color">Color</label>
                <input id="color" value={formData.color} onChange={handleChange} name="color" data-testid="color" placeholder="Color"/>
            <label htmlFor="height">Height</label>
                <input id="height" value={formData.height} onChange={handleChange} name="height" data-testid="height" placeholder="80" />
            <label htmlFor="width">Width</label>
                <input id="width" value={formData.width} onChange={handleChange} name="width" data-testid="width" placeholder="20" />
            <button>Create!</button>
        </form>
    );
}

export default NewBoxForm;