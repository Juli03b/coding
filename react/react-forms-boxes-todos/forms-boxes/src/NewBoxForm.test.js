import { fireEvent, render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("Should render", () => {
    expect( () => render(<NewBoxForm />)).not.toThrow();
});

it("Should match snapshot", () => {
    const {asFragment} = render(<NewBoxForm />);

    expect(asFragment()).toMatchSnapshot();
});

it("Should show text being typed", () => {
    const {getByTestId} = render(<NewBoxForm />);
    const textBox = getByTestId("color");

    fireEvent.change(textBox, {target: {value: "blue"}});
    
    expect(textBox).toHaveDisplayValue("blue");
});