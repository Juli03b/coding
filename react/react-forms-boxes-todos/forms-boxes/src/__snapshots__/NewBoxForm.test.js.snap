// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <form
    data-testid="form"
  >
    <label
      for="color"
    >
      Color
    </label>
    <input
      data-testid="color"
      id="color"
      name="color"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      data-testid="height"
      id="height"
      name="height"
      value=""
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      data-testid="width"
      id="width"
      name="width"
      value=""
    />
    <button>
      Create!
    </button>
  </form>
</DocumentFragment>
`;
